File: java_implementation/src/main/java/com/tambapps/web/page_scrapping/parameters/validator/NotEmptyCollection.java
package com.tambapps.web.page_scrapping.parameters.validator;

import com.beust.jcommander.IParameterValidator;
import com.beust.jcommander.ParameterException;

public class NotEmptyCollection implements IParameterValidator {

  @Override
  public void validate(String name, String value) throws ParameterException {
    if (value.isEmpty()) {
      throw new ParameterException("Parameter " + name + " shouldn't be empty");
    }
  }

}

File: kotlin_implementation/src/main/kotlin/com/tambapps/kotlin/parameters/validator/NotEmptyCollection.kt
package com.tambapps.kotlin.parameters.validator

import com.beust.jcommander.IParameterValidator
import com.beust.jcommander.ParameterException

class NotEmptyCollection: IParameterValidator {
    override fun validate(name: String, value: String) {
        if (value.isEmpty()) {
            throw ParameterException("Parameter $name shouldn't be empty")
        }
    }
}